src/components/widgets/ConstructionSiteWidget.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback, useContext, useEffect } from 'react';
+import { useQuery } from 'react-apollo';
+import { StyleSheet, View } from 'react-native';
+import { NavigationScreenProp } from 'react-navigation';
+
+import { colors, normalize } from '../../config';
+import { ConstructionSiteContext } from '../../ConstructionSiteProvider';
+import { filterForValidConstructionSites, graphqlFetchPolicy } from '../../helpers';
+import { constructionSite } from '../../icons';
+import { NetworkContext } from '../../NetworkProvider';
+import { getQuery, QUERY_TYPES } from '../../queries';
+import { Icon } from '../Icon';
+import { BoldText, RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+import { Wrapper } from '../Wrapper';
+
+type Props = {
+  navigation: NavigationScreenProp<never>;
+};
+
+export const ConstructionSiteWidget = ({ navigation }: Props) => {
+  const { constructionSites, setConstructionSites } = useContext(ConstructionSiteContext);
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp, refreshTime: undefined });
+
+  const { data } = useQuery(getQuery(QUERY_TYPES.PUBLIC_JSON_FILE), {
+    variables: { name: 'constructionSites' },
+    fetchPolicy,
+    skip: !isConnected || !isMainserverUp
+  });
+
+  const onPress = useCallback(() => {
+    if (constructionSites.length) navigation.navigate('ConstructionSiteOverview');
+  }, [constructionSites, navigation]);
+
+  useEffect(() => {
+    if (data) {
+      const constructionSitesPublicJsonFileContent =
+        data.publicJsonFile && JSON.parse(data.publicJsonFile.content);
+
+      setConstructionSites(filterForValidConstructionSites(constructionSitesPublicJsonFileContent));
+    }
+  }, [data, setConstructionSites]);
+
+  return (
+    <Touchable onPress={onPress}>
+      <Wrapper>
+        <View style={styles.container}>
+          <Icon style={styles.icon} xml={constructionSite(colors.primary)} />
+          <BoldText style={styles.count}>{constructionSites.length}</BoldText>
+        </View>
+        <RegularText style={styles.text}>Baustellen</RegularText>
+      </Wrapper>
+    </Touchable>
+  );
+};
+
```

Comment:
for this file the same five points apply like commented in the [event widget PR](https://github.com/ikuseiGmbH/smart-village-app-app/pull/156/files#diff-80a14a3a3ec573140c4b751e0e10b89043deb97bca78ef4da931f08f17030e54)

Reply:
i added a todo.

i will fix it as soon as the event widget is merged to the epic and i updated this branch with it